modules/plugins/languages/scala.nix

```diff
@@ -0,0 +1,141 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib) types optionalString;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.nvim.binds) mkMappingOption;
+  inherit (lib.options) mkOption mkEnableOption;
+  inherit (lib.nvim.dag) entryAfter;
+
+  mkBinding = binding: "vim.api.nvim_buf_set_keymap(bufnr, 'n', '${binding.value}', '<cmd>lua ${binding.action}<CR>', {noremap=true, silent=true, desc='${binding.description}'})";
```

Comment:
Is there a reason we're defining `mkBİnding` here and not using one of the binding functions from lib?

Reply:
Might be just ignorance on my part. lsp/config.nix does a similar thing to set buffer local mappings, which is what I looked at.

Reply:
I've inlined this into `attach_metals_keymaps` now